src/EditorFeatures/Core/FindUsages/ClassifiedSpansAndHighlightSpanFactory.cs

```diff
@@ -44,6 +47,82 @@ internal static class ClassifiedSpansAndHighlightSpanFactory
                 documentSpan.Document, documentSpan.SourceSpan, cancellationToken).ConfigureAwait(false);
         }
 
+        public static async Task<ClassifiedTextElement> BuildClassifiedTextElementForSpansAsync(ImmutableArray<TextSpan> spans, ITextSnapshot snapshot, Document document, CancellationToken cancellationToken)
+        {
+            var classificationService = document.GetLanguageService<IClassificationService>();
+
+            var classifiedSpans = new List<ClassifiedSpan>();
+            var semanticSpans = new List<ClassifiedSpan>();
+
+            foreach (var span in spans)
+            {
+                await classificationService.AddSyntacticClassificationsAsync(document, span, classifiedSpans, cancellationToken).ConfigureAwait(false);
+                await classificationService.AddSemanticClassificationsAsync(document, span, semanticSpans, cancellationToken).ConfigureAwait(false);
+            }
+
+            // replace the spans from SyntacticClassifications with spans from SemanticClassifications
+            classifiedSpans.Sort((a, b) => a.TextSpan.Start.CompareTo(b.TextSpan.Start));
+            semanticSpans.Sort((a, b) => a.TextSpan.Start.CompareTo(b.TextSpan.Start));
+
+            var i = 0;
+            foreach (var semanticSpan in semanticSpans)
+            {
+                var foundMatch = false;
+                for (; i < classifiedSpans.Count; i++)
+                {
+                    if (classifiedSpans[i].TextSpan.Start == semanticSpan.TextSpan.Start
+                        && classifiedSpans[i].TextSpan.Length == semanticSpan.TextSpan.Length)
+                    {
+                        foundMatch = true;
+                        classifiedSpans[i] = semanticSpan;
+                        break;
+                    }
+                }
+
+                // if no match found for the semanticSpan, resets i so the next semanticSpan will start to match from the begining
+                if (!foundMatch)
+                {
+                    i = 0;
+                }
+
+            }
+
+            // Convert spans to textruns
+            var runs = new List<ClassifiedTextRun>();
+            var lastSpanEnd = -1;
+            foreach (var span in classifiedSpans)
+            {
+                // Add whitespace
+                if (lastSpanEnd > 0 && span.TextSpan.Start > lastSpanEnd)
+                {
+                    // find the gap between last span and current span, read the gap text from document
+                    var spanGap = new Span(lastSpanEnd, span.TextSpan.Start - lastSpanEnd);
+                    var spanText = snapshot.GetText(spanGap);
+
+                    // special handling for new line, remove all the whitespace at the begining of the new line before the open brace "{"
+                    if (span.ClassificationType == ClassificationTypeNames.Punctuation)
+                    {
+                        if (spanText.StartsWith("\r\n"))
```

Comment:
You may want to just ask if the span is the end of the line directly.

Reply:
@jasonmalinowski  I cannot find an existing API to check if a span is end of line, could you please point me which API you are referring? Thanks